gateway-messages/Cargo.toml

```diff
@@ -6,11 +6,11 @@ license = "MPL-2.0"
 
 [dependencies]
 bitflags = "1.3.2"
-serde = { version = "1.0.145", default-features = false, features = ["derive"] }
+serde.workspace = true
 serde-big-array = "0.4.1"
 serde_repr = { version = "0.1" }
 static_assertions = "1.1.0"
-uuid = { version = "1.1", default-features = false }
+uuid.workspace = true
```

Comment:
This is a breaking change for `no_std` consumers of `gateway-messages`. Will poke and see what the right combo here is...

Reply:
> Cool! This seems like a great change.

Thanks!

> It seems like in principle, Cargo.lock ought not to have changed. I gather that as part of this, we're unifying some deps that were previously duplicated at slightly different versions, and also updating some.

Yup - I found a few places where versions were inconsistent between crates, and unified them.

> In terms of figuring out which of a crate's features go into which Cargo.toml, is this the approach:
> 
> * find the maximum set of features used by _all_ crates in the workspace and put this into the workspace Cargo.toml
> * for any other features used by any other crate, add those to the crate-specific Cargo.toml

That's right. 

> Is there any tooling to help with that? i.e., if I want to add some feature to some crate, it seems easy to add it to the wrong place.

Nope, at the moment it was done manually. I suspect `cargo tree -e features` could be used here though.

> Did this have any notable change in build time?

Before:
- Clean build: 2m 42s
- Touch nexus/src/lib.rs rebuild: 15.97 s

After:
- Clean build: 2m 41s
- Touch nexus/src/lib.rs rebuild: 14.12 s

So... maybe *marginally*, but it's kinda hard to tell.

Reply:
I believe this has been fixed with the new way we're depending on UUID, as well as the cargo features. Is this working well enough for your use-case?

Reply:
Yep. I plan to pull out this crate (and maybe a bit more) soon to avoid the hazard of having a single no_std crate lurking here, but for now the UUID change + cargo feature keeps things working. Thanks!